QgisModelBaker/gui/export.py

```diff
@@ -70,23 +68,24 @@ def __init__(self, tool_name, uri, schema=None):
 
         modelnames = list()
 
-        try:
-            if tool_name == 'ili2gpkg':
-                self._db_connector = gpkg_connector.GPKGConnector(uri, None)
-            else:
-                self._db_connector = pg_connector.PGConnector(uri, schema)
-
-            if self._db_connector.db_or_schema_exists():
-                db_models = self._db_connector.get_models()
-                for db_model in db_models:
-                    regex = re.compile(r'(?:\{[^\}]*\}|\s)')
-                    for modelname in regex.split(db_model['modelname']):
-                        if modelname and modelname not in blacklist:
-                            modelnames.append(modelname.strip())
-        except:
-            # when wrong connection parameters entered, there should just be returned an empty model - so let it pass
-            # The exception can be a lot of different things (depending on the backend) so let's catch all
-            pass
+        if tool:
+            try:
+                db_simple_factory = DbSimpleFactory()
+                db_factory = db_simple_factory.create_factory(tool)
+                self._db_connector = db_factory.get_db_connector(uri, schema)
+                
+                if self._db_connector.db_or_schema_exists():
+                    db_models = self._db_connector.get_models()
+                    for db_model in db_models:
+                        regex = re.compile(r'(?:\{[^\}]*\}|\s)')
+                        for modelname in regex.split(db_model['modelname']):
+                            if modelname and modelname not in blacklist:
+                                modelnames.append(modelname.strip())
+            except:
+                # when wrong connection parameters entered, there should just be returned an empty model - so let it pass
+                # The exception can be a lot of different things (depending on the backend) so let's catch all
```

Comment:
The models in the database are queried each time the connexion parameters change and many times the parameters are wrong. Users are not notified because they would see many warning message. However, I added DBConnectorError type  to the caught exception to limit it to connection errors.